src/charm.py

```diff
@@ -74,10 +89,14 @@ def _on_config_changed(self, event):
         # new config file to the workload container
         prometheus_config = self._prometheus_config()
         config_hash = str(hashlib.md5(str(prometheus_config).encode('utf-8')))
-        if not self._stored.prometheus_config_hash == config_hash:
-            self._stored.prometheus_config_hash = config_hash
-            container.push(PROMETHEUS_CONFIG, prometheus_config)
-            logger.info("Pushed new configuration")
+        if self._stored.prometheus_config_hash != config_hash:
+            try:
+                container.push(PROMETHEUS_CONFIG, prometheus_config)
+                self._stored.prometheus_config_hash = config_hash
+                logger.info("Pushed new configuration")
+            except ConnectionError:
+                logger.info("Ignoring config changed since pebble is not ready")
+                return
```

Comment:
this is an anti pattern imho: `_stored.pebble_ready` seems like a better approach.

Reply:
Why do you think it is the "better approach" ? .  I do not think charms should be tracking the history of Juju events and storing this in their state.

Reply:
With`_stored.pebble_ready` you'd have a one-liner assignment, and elsewhere you'd have a one-liner check.
The try-except block here is effectively tracking the history of juju events as well :)